reconcile/status_board.py

```diff
@@ -281,21 +276,25 @@ def get_current_products_applications_services(
                 logging.error(f'Product "{p.name}" has no id')
                 continue
             p.applications = [
-                Application(**a) for a in get_product_applications(ocm_api, p.id)
+                Application(**a, product=p)
+                for a in get_product_applications(ocm_api, p.id)
             ]
             for a in p.applications:
                 if not a.id:
                     logging.error(f'Application "{a.name}" has no id')
                     continue
                 a.services = [
-                    Service(**s) for s in get_application_services(ocm_api, a.id)
+                    Service(**s, application=a)
+                    for s in get_application_services(ocm_api, a.id)
                 ]
 
         return products
 
     @staticmethod
     def desired_abstract_status_board_map(
-        desired_product_apps: Mapping[str, set[str]], slodocs: list[SLODocumentV1]
+        current_status_board_map: Mapping[str, AbstractStatusBoard],
```

Comment:
this will make desired state depends on current state to construct, while these id fields are only needed in diff and reconcile, how about keep these id as None for desired state, but sync them during diff in one go for all classes of `AbstractStatusBoard`, like this:


```python
        for pair in chain(diff_result.identical.values(), diff_result.change.values()):
            pair.desired.id = pair.current.id
```

Reply:
Neat. Done.